File: java/binarysearchtree/Node.java
class Node {

    public Node left; // public for the lazy
    public Node right;
    
    private int value;
    
    public Node(int value, Node left, Node right) {
        this.value = value;
        this.left = left;
        this.right = right;
    }

    public int getValue() {
        return value;
    }

    public void insertChild(int val) {
        if (value > val) {
            if (left == null) {
                left = new Node(val, null, null);
            } else {
                left.insertChild(val);
            }
        } else {
            if (right == null) {
                right = new Node(val, null, null);
            } else {
                right.insertChild(val);
            }
        }
    }
}

File: python/datastructures/binarysearchtree/node.py
#!/usr/bin/env python3

"""
Node object
"""

class Node(object):

    def __init__(self, value, left=None, right=None):
        self.left = left
        self.right = right
        self.value = value

    def __str__(self):
        return "Value %s" % self.value;

    def insert_child(self, value):
        if self.value > value:
            if self.left is None:
                self.left = Node(value)
            else:
                self.left.insert_child(value)
        else:
            if self.right is None:
                self.right = Node(value)
            else:
                self.right.insert_child(value)
